pytorch_lightning/callbacks/pruning.py

```diff
@@ -0,0 +1,316 @@
+                raise MisconfigurationException(
+                    f"The provided parameter_names {param_name} isn't in {self.PARAMETER_NAMES} "
+                )
+
+        if isinstance(pruning_fn, str):
+            pruning_fn = pruning_fn.lower()
+            if pruning_fn not in _PYTORCH_PRUNING_FUNCTIONS:
+                raise MisconfigurationException(
+                    f"The provided pruning_fn {pruning_fn} isn't available with "
+                    f"PyTorch build-in {_PYTORCH_PRUNING_FUNCTIONS.keys()} "
+                )
+            if "unstructured" not in pruning_fn:
+                if pruning_dim is None:
+                    raise MisconfigurationException(
+                        "When requesting `structured` pruning, the `pruning_dim` should be provided."
+                    )
+                if pruning_fn == "ln_structured":
+                    if pruning_norm is None:
+                        raise MisconfigurationException(
+                            "When requesting `ln_structured` pruning, the `pruning_norm` should be provided."
+                        )
+
+                    pruning_fn = self.create_pruning_fn(pruning_fn, dim=pruning_dim, n=pruning_norm)
+            else:
+                pruning_fn = self.create_pruning_fn(pruning_fn)
+
+        self.pruning_fn = pruning_fn
+
+        if not (prune_on_epoch_end or prune_on_fit_end):
+            raise MisconfigurationException(
+                "The `ModelPruning` won't be triggered as currently not activated on either `epoch_end` or `fit_end`.")
+
+        self.make_pruning_permanent = make_pruning_permanent
+
+        if not isinstance(amount, (int, float, Callable)):
+            raise MisconfigurationException(
+                "amount should be provided and be either an int, a float or Callable function."
+            )
+
+        self.amount = amount
+
+    def create_pruning_fn(self, pruning_fn: str, *args, **kwargs):
+        if self.use_global_unstructured:
+            pruning_fn = _PYTORCH_PRUNING_METHOD[pruning_fn]
+            self._global_kwargs = kwargs
+            return pruning_fn
+        else:
+            return ModelPruning._wrap_pruning_fn(_PYTORCH_PRUNING_FUNCTIONS[pruning_fn], **kwargs)
+
+    @staticmethod
+    def _wrap_pruning_fn(pruning_fn, *args, **kwargs):
+        return lambda module, name, amount: pruning_fn(
+            module, name, amount, *args, **kwargs
+        )
+
+    def _make_pruning_permanent(self):
+        for module, param_name in self.parameters_to_prune:
+            pytorch_prune.remove(module, param_name)
+
+    def _resolve_amout(self, current_epoch: int) -> float:
+        if isinstance(self.amount, Callable):
+            amount_fn = self.amount
+            amount = amount_fn(current_epoch)
+        else:
+            amount = self.amount
+        return amount
+
+    def _apply_lottery_ticket_hypothesis(self, module: nn.Module, orig_module: nn.Module, tensor_name: str):
```

Comment:
to clarify maybe just do `_restore_original_weights`

Reply:
Done !